corelib/crayfish_renderer.cpp

```diff
@@ -297,264 +311,343 @@ void Renderer::drawVectorData(const Output* output)
-  double sinAlpha = sin( vectorAngle ) * mag(xVal);
-
-  // Now determine the X and Y distances of the end of the line from the start
-  float xDist = 0.0;
-  float yDist = 0.0;
-  switch (mCfg.ds.mShaftLengthMethod)
-  {
+
+    if (cfg->ds.mVectorFilterMin >= 0 && magnitude < cfg->ds.mVectorFilterMin)
+        return true;
+    if (cfg->ds.mVectorFilterMax >= 0 && magnitude > cfg->ds.mVectorFilterMax)
+        return true;
+
+    // Determine the angle of the vector, counter-clockwise, from east
+    // (and associated trigs)
+    double vectorAngle = -1.0 * atan( (-1.0 * yVal) / xVal );
+    cosAlpha = cos( vectorAngle ) * mag(xVal);
+    sinAlpha = sin( vectorAngle ) * mag(xVal);
+
+    // Now determine the X and Y distances of the end of the line from the start
+    float xDist = 0.0;
+    float yDist = 0.0;
+    switch (cfg->ds.mShaftLengthMethod)
+    {
     case ConfigDataSet::MinMax:
     {
-      float minShaftLength = mCfg.ds.mMinShaftLength;
-      float maxShaftLength = mCfg.ds.mMaxShaftLength;
-      float minVal = output->dataSet->minZValue();
-      float maxVal = output->dataSet->maxZValue();
-      double k = (V - minVal) / (maxVal - minVal);
-      double L = minShaftLength + k * (maxShaftLength - minShaftLength);
-      xDist = cosAlpha * L;
-      yDist = sinAlpha * L;
-      break;
+        float minShaftLength = cfg->ds.mMinShaftLength;
+        float maxShaftLength = cfg->ds.mMaxShaftLength;
+        float minVal = output->dataSet->minZValue();
+        float maxVal = output->dataSet->maxZValue();
+        double k = (magnitude - minVal) / (maxVal - minVal);
+        double L = minShaftLength + k * (maxShaftLength - minShaftLength);
+        xDist = cosAlpha * L;
+        yDist = sinAlpha * L;
+        break;
     }
     case ConfigDataSet::Scaled:
     {
-      float scaleFactor = mCfg.ds.mScaleFactor;
-      xDist = scaleFactor * xVal;
-      yDist = scaleFactor * yVal;
-      break;
+        float scaleFactor = cfg->ds.mScaleFactor;
+        xDist = scaleFactor * xVal;
+        yDist = scaleFactor * yVal;
+        break;
     }
     case ConfigDataSet::Fixed:
     {
-      // We must be using a fixed length
-      float fixedShaftLength = mCfg.ds.mFixedShaftLength;
-      xDist = cosAlpha * fixedShaftLength;
-      yDist = sinAlpha * fixedShaftLength;
-      break;
+        // We must be using a fixed length
+        float fixedShaftLength = cfg->ds.mFixedShaftLength;
+        xDist = cosAlpha * fixedShaftLength;
+        yDist = sinAlpha * fixedShaftLength;
+        break;
+    }
+    }
+
+    // Flip the Y axis (pixel vs real-world axis)
+    yDist *= -1.0;
+
+    if (qAbs(xDist) < 1 && qAbs(yDist) < 1)
+        return true;
+
+    // Determine the line coords
+    lineEnd = QPointF( lineStart.x() + xDist,
+                       lineStart.y() + yDist);
+
+    vectorLength = sqrt(xDist*xDist + yDist*yDist);
+
+    // Check to see if any of the coords are outside the QImage area, if so, skip the whole vector
```

Comment:
How about vectors that are partially visible - we want to render those too, right?